Visitor/Arithmetic.php

```diff
@@ -185,7 +185,7 @@ public function visit(
                 if (null  === $parent ||
                     $type === $parent->getId()) {
                     $acc = function ($b) use ($a, $acc) {
-                        if (0 === $b) {
+                        if (0.0 === floatval($b)) {
```

Comment:
Actually, I am not sure if we can something different than a float here. I am pretty sure we always have a float, so the cast is not needed.